packages/sdk/src/sdk/api/events/EventsApi.ts

```diff
@@ -0,0 +1,108 @@
+import snakecaseKeys from 'snakecase-keys'
+
+import { LoggerService } from '../../services'
+import {
+  EntityManagerService,
+  EntityType,
+  Action
+} from '../../services/EntityManager/types'
+import { decodeHashId } from '../../utils/hashId'
+import {
+  Configuration,
+  EventsApi as GeneratedEventsApi
+} from '../generated/default'
+
+type EventMetadata = {
+  userId: number
+  eventId: number
+  eventType?: string
+  entityType?: string
+  entityId?: number
+  isDeleted?: boolean
+  endDate?: string // ISO format date string
```

Comment:
* This file needs zod validation for inputs to functions
* We could just accept a date for this field so we maintain control over how it gets serialized. That or we set up the zod validator to only allow that format and throw otherwise
* Might want to consider string unions ('track' | 'playlist' | ...) or enums for eventType and entityType, or pull in the types from the responses in the generated API so that this stays in sync automatically.